Sources/Testing/EntryPoints/EntryPoint.swift

```diff
@@ -220,12 +220,12 @@ public struct __CommandLineArguments_v0: Sendable {
   /// The version of the event stream schema to use when writing events to
   /// ``experimentalEventStreamOutput``.
   ///
-  /// If the value of this property is `nil`, events are encoded verbatim (using
-  /// ``Event/Snapshot``.) Otherwise, the corresponding stable schema is used
-  /// (e.g. ``ABIv0/Record`` for `0`.)
+  /// The corresponding stable schema is used to encode events to the event
+  /// stream (for example, ``ABIv0/Record`` is used if the value of this
+  /// property is `0`.)
   ///
-  /// - Warning: The behavior of this property will change when the ABI version
-  ///   0 JSON schema is finalized.
+  /// If the value of this property is `nil`, the testing library assumes a
+  /// value of `0` instead.
```

Comment:
I'd say it should assume whatever is the _newest_ (i.e. latest) version at the time it is called, if nil. Right now that is 0, but will presumably increment as subsequent versions are added

Reply:
Yeah, it's a classic problem with any versioning scheme: what should the default be if no version is explicitly specified? If you default to oldest, newcomers may not be aware there is a newer version they can be using and that newer version may struggle to gain traction among the user base. But if you default to latest, a newer version may break older users who weren't explicitly passing any version and assuming an older one.

Two thoughts about this:

1. When it comes to an **entry point _function_**, such as `swt_abiv0_getEntryPoint`, the version is already specified explicitly in the symbol name, so there will never be a scenario where it's not explicit. Good! That avoids that class of problems (…right?).

    Although that raises a different question: what if someone called that function but passed a different version in the `__CommandLineArguments_v0` struct property?
2. When it comes to **command-line users of SwiftPM**, they might not pass the new `--event-stream-version` flag at all, so that's the scenario where we need to decide what the default should be.

It's common for Swift packages to predicate breaking behavioral changes on the `// swift-tools-version: <vers>` line in the package's manifest file. To solve problem 2 above, what if we planned to do this as well? SwiftPM could be modified such that, if no version flag is explicitly passed, it passes the version which was known to be latest at the time when the package's tools version was initially released. Then, it would always pass the version flag to the runner process, so the runner always receives an explicit version.

Reply:
> Although that raises a different question: what if someone called that function but passed a different version in the __CommandLineArguments_v0 struct property?

This is going to be moot eventually because `__CommandLineArguments_v0` will be replaced by an "ABI stable" JSON format, same as how the snapshot types are being replaced. We just haven't gotten to that point yet. Once that's done, there will probably be a mandatory  `"version": 0` field in the JSON blob, and we'll teach our JSON decoding code to try a few different formats before throwing up.

Reply:
> To solve problem 2 above, what if we planned to do this as well? SwiftPM could be modified such that, if no version flag is explicitly passed, it passes the version which was known to be latest at the time when the package's tools version was initially released. Then, it would always pass the version flag to the runner process, so the runner always receives an explicit version.

This way lies danger for tools though. If I write a tool that doesn't pass `--event-stream-version` today, and I build somebody else's package that says `// swift-tools-version: 7.0`, I am not expecting to get the version 2 (or whatever) of the JSON format. My code was written against what was current when I wrote it. The problem here is that the package being built and tested may have no relationship to the tool that's reading the JSON output.

It's better to default to the lowest common denominator because it's stable in the long term. We can/should advise tools authors to be explicit about their requirements, but having it spontaneously change formats is asking for trouble down the line.

Reply:
Another alternative would be to mandate that if you pass the flag requesting the JSON output, you must pass the flag specifying a version explicitly too.

But that being said, I honestly think it may be best for us to just document that we default to _newest_, and strongly recommend that tools consuming this info via the SwiftPM CLI explicitly pass the version flag (but not mandate it). This decision can be aligned with the toolchain version once swift-testing is part of toolchains, so that means it would only potentially break if/when a newer toolchain is installed, and there's already some precedent for downstream tools needing to be adjusted to accommodate new toolchains.

Reply:
Discussed further and we'll have `nil` equal "whatever the newest version is" and advise tools authors to always be explicit.